packages/popover/test/position.test.js

```diff
@@ -171,4 +172,300 @@ describe('position', () => {
       });
     });
   });
+
+  describe('viewport constraint', () => {
+    let constraintPopover, constraintTarget, constraintOverlay;
+
+    beforeEach(async () => {
+      constraintPopover = fixtureSync('<vaadin-popover></vaadin-popover>');
+      constraintPopover.renderer = (root) => {
+        if (!root.firstChild) {
+          const div = document.createElement('div');
+          div.textContent = 'This is a long popover content that will extend beyond edges';
+          root.appendChild(div);
+        }
+      };
+      await nextRender();
+      constraintOverlay = constraintPopover.shadowRoot.querySelector('vaadin-popover-overlay');
+    });
+
+    async function openAndMeasure() {
+      constraintPopover.opened = true;
+      await oneEvent(constraintOverlay, 'vaadin-overlay-open');
+      const overlayRect = constraintOverlay.$.overlay.getBoundingClientRect();
+      const targetRect = constraintTarget.getBoundingClientRect();
+      return { overlayRect, targetRect };
+    }
+
+    describe('bottom position near left edge', () => {
```

Comment:
Looks like the test suites for top and bottom are basically the same, except for changing the popover position? If so we could reduce some duplication with a loop: 
```
['top', 'bottom'].forEach(position => describe(`${position} position near left edge`) ...)
```

This would also be easy to add to the RTL suite to cover both positions.